src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,57 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFromFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder firstFileData = new StringBuilder();
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName))) {
+            String value = br.readLine();
+            while (value != null) {
+                firstFileData.append(value).append("|");
+                value = br.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return firstFileData.toString();
+    }
+
+    private String createReport(String dataFromFile) {
+        StringBuilder report = new StringBuilder();
+        int supply = 0;
+        int buy = 0;
+        for (String str : dataFromFile.split("\\|")) {
+            String[] a = str.split(",");
+            if (a[0].equals("buy")) {
+                buy += Integer.parseInt(a[1]);
+            } else {
+                supply += Integer.parseInt(a[1]);
+            }
```

Comment:
```suggestion
        for (String str : dataFromFile.split("\\|")) {
            String[] splitted = str.split(",");
            if (a[0].equals("buy")) {
                buy += Integer.parseInt(a[1]);
            } else {
                supply += Integer.parseInt(a[1]);
            }
```
add constants for indexes